lib/AST/ClangTypeConverter.cpp

```diff
@@ -118,23 +118,34 @@ const clang::Type *ClangTypeConverter::getFunctionType(
     ArrayRef<AnyFunctionType::Param> params, Type resultTy,
     AnyFunctionType::Representation repr) {
 
-  SmallVector<Type, 4> paramsTy;
-  for (auto p : params)
-    paramsTy.push_back(p.getPlainType());
+#define HANDLE_NULL_AND_FUNCTION_TYPE(ty)            \
+  if (ty.isNull())                                   \
+    return nullptr;                                  \
+  if (auto *fnTy = ty->getAs<clang::FunctionType>()) \
+    ty = ClangASTContext.getPointerType(ty);
 
   auto resultClangTy = convert(resultTy);
-  if (resultClangTy.isNull())
-    return nullptr;
-
-  SmallVector<clang::QualType, 8> paramsClangTy;
-  for (auto p : paramsTy) {
-    auto pc = convert(p);
-    if (pc.isNull())
-      return nullptr;
+  HANDLE_NULL_AND_FUNCTION_TYPE(resultClangTy);
```

Comment:
Hmm.  General type conversion should turn a `@convention(c)` function type into a function pointer type, and maybe there should be a way to request that to be suppressed when doing it at the top level.  It shouldn't be necessary to handle function types manually in every recursive position they might appear.

Reply:
Yeah, that's true. For some reason, we are not handling converting `@convention(c)` functions to have that pointer, but we do have that for blocks. I must've made a mistake when porting the code from `GenClangType`, which does have that.

Will fix and re-push.